data_validation/data_validation.py

```diff
@@ -216,6 +215,8 @@ def execute_recursive_validation(self, validation_builder, grouped_fields):
                     )
         elif self.config_manager.primary_keys:
             validation_builder.add_config_query_groups(self.config_manager.primary_keys)
+
+        elif self.config_manager.primary_keys and len(grouped_fields) == 0:
```

Comment:
Possible the order of these if statements is wrong. I don't think this branch will ever be reached since `elif self.config_manager.primary_keys:` appears above it.

Reply:
I don't believe primary keys check below can ever be reached since the grouped_fields value here will always be ==0 at this point